src/config_listener_mon.erl

```diff
@@ -0,0 +1,84 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(config_listener_mon).
+-behaviour(gen_server).
+-vsn(1).
+
+
+-export([
+    start_link/2
+]).
+
+
+-export([
+    init/1,
+    terminate/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/3
+]).
+
+
+-record(st, {
+    pid,
+    ref
+}).
+
+
+start_link(Module, InitSt) ->
+    proc_lib:start(?MODULE, init, [{self(), Module, InitSt}]).
+
+
+init({Pid, Mod, InitSt}) ->
+    Ref = erlang:monitor(process, Pid),
+    case config_listener:start(Mod, {Mod, Pid}, {Pid, InitSt}) of
```

Comment:
I agree that it leaks internals but it doesn't change anything on the current behavior. I had to do this to match the old behavior where we were passing {Subscriber, InitState} with the gen_event handler so that the Subscriber pid could be passed to handle_config_terminate (weirdly). I just moved it along with the call to config_listener:start/3 rather than keeping it wrapped in config.erl's call to config_listener_mon:start_link/2.